src/spyglass/position/position_position.py

```diff
@@ -0,0 +1,542 @@
+)
+from .dlc_utils import get_video_path, check_videofile
+from .position_dlc_selection import DLCPos
+from .position_trodes_position import TrodesPos
+
+schema = dj.schema("position_position")
+
+
+@schema
+class PosSource(dj.Manual):
+    """
+    Table to identify source of Position Information from upstream options
+    (e.g. DLC, Trodes, etc...) To add another upstream option, a new Part table
+    should be added in the same syntax as DLCPos and TrodesPos and
+    PosSelect source header should be modified to include the name.
+    """
+
+    definition = """
+    -> IntervalList
+    source: enum("DLC", "Trodes")
+    position_id: int
+    ---
+    """
+
+    class DLCPos(dj.Part):
+        """
+        Table to pass-through upstream DLC Pose Estimation information
+        """
+
+        definition = """
+        -> PosSource
+        -> DLCPos
+        ---
+        -> AnalysisNwbfile
+        position_object_id : varchar(80)
+        orientation_object_id : varchar(80)
+        velocity_object_id : varchar(80)
+        """
+
+    class TrodesPos(dj.Part):
+        """
+        Table to pass-through upstream Trodes Position Tracking information
+        """
+
+        definition = """
+        -> PosSource
+        -> TrodesPos
+        ---
+        -> AnalysisNwbfile
+        position_object_id : varchar(80)
+        orientation_object_id : varchar(80)
+        velocity_object_id : varchar(80)
+        """
+
+    def insert1(self, key, params: Dict = None, **kwargs):
+        position_id = key.get("position_id", None)
+        if position_id is None:
+            key["position_id"] = (
+                dj.U().aggr(self & key, n="max(position_id)").fetch1("n") or 0
+            ) + 1
+        else:
+            id = (self & key).fetch("position_id")
+            if len(id) > 0:
+                position_id = max(id) + 1
+            else:
+                position_id = max(0, position_id)
+            key["position_id"] = position_id
+        super().insert1(key, **kwargs)
+        source = key["source"]
+        part_table = getattr(self, f"{source}Pos")
+        table_query = (
+            dj.FreeTable(dj.conn(), full_table_name=part_table.parents()[1]) & params
+        )
+        (
+            analysis_file_name,
+            position_object_id,
+            orientation_object_id,
+            velocity_object_id,
+        ) = table_query.fetch1(
+            "analysis_file_name",
+            "position_object_id",
+            "orientation_object_id",
+            "velocity_object_id",
+        )
+        part_table.insert1(
+            {
+                **key,
+                "analysis_file_name": analysis_file_name,
+                "position_object_id": position_object_id,
+                "orientation_object_id": orientation_object_id,
+                "velocity_object_id": velocity_object_id,
+                **params,
+            },
+        )
+
+
+@schema
+class IntervalPositionInfoSelection(dj.Manual):
```

Comment:
And also think this should be renamed as something other than IntervalPositionInfo so we can import IntervalPositionInfo from common_position without conflict (and avoiding `import as common_IntervalPositionInfo`) @edeno @khl02007.